### PR TITLE
build: asserting files and contents in tests only executes container …

### PR DESCRIPTION
**Pull request by jazzdan**
…once

### CODE REVIEW

File: internal/build/docker_test.go [hunk 1]
```diff
@@ -35,26 +36,6 @@ func TestDigestFromSingleStepOutput(t *testing.T) {
 	}
 }
 
-func TestBuildBase(t *testing.T) {
```
Comment (jazzdan) [inline internal/build/docker_test.go]: We don't really need this test any more, it's entirely covered by the other test.

File: internal/build/docker_test.go [hunk 2]
```diff
@@ -132,11 +119,22 @@ func (f *testFixture) newBuilderForTesting() *localDockerBuilder {
...
+type pathContent struct {
+	path     string
+	contents string
+}
+
+func (f *testFixture) assertFilesInImageWithContents(tag digest.Digest, contents []pathContent) {
 	ctx := context.Background()
+	var cmd strings.Builder
+	for _, c := range contents {
+		cs := fmt.Sprintf("cat %s | grep \"%s\" || echo \"contents not found\"; ", c.path, c.contents)
```
Comment (maiamcc) [inline internal/build/docker_test.go]: hmm do we want to differentiate the error message between file not found and contents not found? (at very least, err should read `file not found or has wrong contents`). or can you maybe printf the file name or the bad contents into the `echo`? this doesn't give the user much debug info...

File: internal/build/docker_test.go [hunk 3]
```diff
@@ -167,10 +165,10 @@ func (f *testFixture) assertFileInImageWithContents(tag digest.Digest, path stri
 	io.Copy(output, out)
 
 	if strings.Contains(output.String(), "No such file") {
-		f.t.Errorf("Expected to find file %s in container. Got: %s", path, output)
+		f.t.Errorf("Expected to find file in container. Got: %s", output)
```
Comment (maiamcc) [inline internal/build/docker_test.go]: think this conditional is useless now: this is the bad output from `cat`, and `grep` will eat the output of `cat`, right?

File: internal/build/docker_test.go [hunk 4]
```diff
@@ -167,10 +165,10 @@ func (f *testFixture) assertFileInImageWithContents(tag digest.Digest, path stri
...
 	}
 
-	if !strings.Contains(output.String(), contents) {
-		f.t.Errorf("Expected file %s to have contents %s but got %s", path, contents, output)
+	if strings.Contains(output.String(), "contents not found") {
+		f.t.Errorf("Expected to find contents in container. Got: %s", output)
```
Comment (maiamcc) [inline internal/build/docker_test.go]: maybe: `Failed to find one or more expected files in container with output: %s`

File: internal/build/docker_test.go [hunk 5]
```diff
@@ -74,9 +58,12 @@ func TestMount(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// TODO(dmiller): this is slow
-	f.assertFileInImageWithContents(tag, "/src/hi/hello", "hi hello")
-	f.assertFileInImageWithContents(tag, "/src/sup", "my name is dan")
+	pcs := []pathContent{
```
Comment (maiamcc) [inline internal/build/docker_test.go]: hmm it feeeeels like we should a single struct that holds test data that we can both write files from and assert test behavior from...? any thoughts? seems awk to rewrite the same string multiple times :-/

File: internal/build/docker_test.go (hunk 2 above)
Comment (jazzdan) [reply internal/build/docker_test.go]: I couldn't think of a way to do this, but in my experience getting the stdout printed to you is enough debug info.

PRs welcome! :)

File: internal/build/docker_test.go (hunk 5 above)
Comment (jazzdan) [reply internal/build/docker_test.go]: yeah this would be better if we were doing table tests...

File: internal/build/docker_test.go (hunk 2 above)
Comment (maiamcc) [reply internal/build/docker_test.go]: ` echo \"could not find file %s / did not have contents %s\"` and sprintf it in

**Pull request merged**